Library/Homebrew/cask/audit.rb

```diff
@@ -282,6 +284,66 @@ def check_token_conflicts
       add_warning "possible duplicate, cask token conflicts with Homebrew core formula: #{core_formula_url}"
     end
 
+    def check_token_valid
+      return unless @strict
+
+      add_warning "cask token is not lowercase" if cask.token.downcase!
+
+      add_warning "cask token contains non-ascii characters" unless cask.token.ascii_only?
+
+      add_warning "cask token + should be replaced by -plus-" if cask.token.include? "+"
+
+      add_warning "cask token @ should be replaced by -at-" if cask.token.include? "@"
+
+      add_warning "cask token whitespace should be replaced by hyphens" if cask.token.include? " "
+
+      add_warning "cask token underscores should be replaced by hyphens" if cask.token.include? "_"
+
+      if cask.token.match?(/[^a-z0-9\-]/)
+        add_warning "cask token should only contain alphanumeric characters and hyphens"
+      end
+
+      add_warning "cask token should not contain double hyphens" if cask.token.include? "--"
+
+      return unless cask.token.end_with?("-") || cask.token.start_with?("-")
+
+      add_warning "cask token should not have leading or trailing hyphens"
+    end
+
+    def check_token_bad_words
+      return unless @strict
+
+      add_warning "cask token contains .app" if cask.token.end_with? ".app"
+
+      if cask.token.end_with? "alpha", "beta", "release candidate"
+        add_warning "cask token contains version designation"
+      end
+
+      add_warning "cask token mentions launcher" if cask.token.end_with? "launcher"
+
+      add_warning "cask token mentions desktop" if cask.token.end_with? "desktop"
+
+      add_warning "cask token mentions platform" if cask.token.end_with? "mac", "osx"
+
+      add_warning "cask token mentions architecture" if cask.token.end_with? "x86", "32-bit", "ppc"
```

Comment:
`x86_64` and `64-bit` also occur. `ppc` can be excluded, since Homebrew itself doesn’t support it.